### ISSUE TITLE
SD card logging error

### ISSUE THREAD

**Issue by Rohan5534** [Rohan5534/diy-power-monitor#1]
Writing to SD card randomly stops after some time. 

Minimum time before logging may produce errors is 30 minutes. i.e. the logging works fine for at least 25 minutes.

After that, the logging stops at random time (sometimes 1600 Sec since startup , 2000 sec since startup , 1800 sec since startup and so on). 
Maximum time for which logging worked was about 6700 secs.

No idea what might be causing error. 

Currently I have made a simple circuit using just arduino and sd card. It is logging comparable values into the SD card at same time interval. Will update with results ASAP.

**Comment by Rohan5534** [comment]
UPDATE : I tested just the SD card with arduino writing at same speed and it works without a problem.

**Comment by Rohan5534** [comment]
I think I have resolved the problem. 

I think it was a bug in the code. 

To log a value to SD card at each second, the code uses timer interrupts at 100 msec, and every time the interrupt is called, it increaments count variable. And when the count reaches 10, we know that it has been 1 sec and we log the value and set count again to 0. 

The problem is that in code , I was checking `if(count ==10) {LOG TO SD CARD;}`. But it might have been that while running code , sometimes the count might get incremented from 9 to 10 to 11 and maybe in that period, the control would not have gotten to the logging condition. And once the count went to 11, there was no way for it to reset to 0 , so it just went on increasing and caused the error. 

Hence , I changed the condition from `if(count == 10)` to `if(count > 9)`. This way even if we miss a second, there is a way for the count to reset to zero. 

I have not fully tested the solution , but it is working so far. And i think this was it , hence i am closing the issue , will reopen if error happens again.
